test/e2e/kubevirt.go

```diff
@@ -303,8 +312,8 @@ var _ = Describe("Kubevirt Virtual Machines", func() {
 		checkPodRunningReady = func() func(Gomega, *corev1.Pod) {
 			return func(g Gomega, pod *corev1.Pod) {
 				ok, err := testutils.PodRunningReady(pod)
-				Expect(err).ToNot(HaveOccurred())
-				Expect(ok).To(BeTrue())
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(ok).To(BeTrue())
```

Comment:
these changes aren't needed.

Reply:
They are, in fact is a bug we have at the test suite, when we use this functionat at Eventually it will not retry if it fails (it did happend to me).

Reply:
I honestly don't understand how these functions that receive gomega work / what problem that solves / why isn't this code simpler. 

Using expect inside an eventually is simply something we should stay away from. Honestly, all expects inside functions IMHO.

Reply:
It's all related to this 

```
Eventually also supports a special case polling function that takes a single Gomega argument and returns no values. Eventually assumes such a function is making assertions and is designed to work with the Succeed matcher to validate that all assertions have passed.

For example:

Eventually(func(g Gomega) {
   model, err := client.Find(1138)
   g.Expect(err).NotTo(HaveOccurred())
   g.Expect(model.Reticulate()).To(Succeed())
   g.Expect(model.IsReticulated()).To(BeTrue())
   g.Expect(model.Save()).To(Succeed())
}).Should(Succeed())
will rerun the function until all assertions pass.
```

Reply:
I can't prevent you from using a gomega feature.

I prefer the simple alternative of having the fn return an error, and that it eventually / consistently succeeds - but that's opinionated.

Thanks for explaining to me this new feature.